### PR TITLE
[2.2.x]Fix monitoring pvc issue

### PR DESCRIPTION
**Pull request by loganhz**
Proposed changes
======
Fix monitoring pvc issue
Backend will add a new param `prometheus.persistent.useReleaseName` for backward compatibility.

Note:

This PR can be merged anytime as it won't break anything. But we cannot validate it until backend change is in. 

Types of changes
======
- Bugfix (non-breaking change which fixes an issue)

Linked Issues
======
https://github.com/rancher/rancher/issues/19610

 

Further comments
======

### CODE REVIEW

File: lib/monitoring/addon/components/enable-monitoring/component.js [hunk 1]
```diff
@@ -94,6 +94,9 @@ export default Component.extend(ReservationCheck, {
       answers['prometheus.resources.core.limits.cpu'] = `${ get(this, 'limitsCpu') }m`;
       answers['prometheus.resources.core.requests.memory'] = `${ get(this, 'requestsMemory') }Mi`;
       answers['prometheus.resources.core.limits.memory'] = `${ get(this, 'limitsMemory') }Mi`;
+      if ( !get(this, 'enabled') || get(this, 'useReleaseName') ) {
+        answers['prometheus.persistent.useReleaseName'] = 'true';
```
Comment (westlywright) [inline lib/monitoring/addon/components/enable-monitoring/component.js]: Is there a reason we're using string over bool here?

Comment (loganhz) [reply lib/monitoring/addon/components/enable-monitoring/component.js]: Backend only supports string for now.

**Comment by loganhz** [comment]
backend in

**Comment by cipianpascu** [comment]
Hello,

In version 2.2.6 I still get this issue. More than that, once enabled, monitoring can not be disabled anymore. I pressed the disable button, I manually removed the helm carts from the app list, but it is in vain.

Ciprian

**Pull request merged**